internal/website/gatherexamples/gatherexamples.go

```diff
@@ -0,0 +1,301 @@
+// Copyright 2019 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Command gatherexamples extracts examples in a Go module into a JSON-formatted
+// object. This is used as input for building the Go CDK Hugo website.
+// Examples must include a comment that starts with "// This example is used in"
+// somewhere in the function body in order to be included in this tool's output.
+//
+// gatherexamples does some minimal rewriting of the example source code for
+// presentation:
+//
+//   - Any imports the example uses will be prepended to the code.
+//   - log.Fatal(err) -> return err
+//   - A comment line "// Variables set up elsewhere:" will remove any code up
```

Comment:
(Missed this one, sorry.)

Using `context.Background()` is usually the wrong choice in most code, so keeping it in for godoc examples is necessary but annoying. Opening up portable types usually requires some amount of code that's unrelated to the actual action, so it would be easier to write these as `var bucket *blob.Bucket`, which is just noise for the final example.